modules/aggregation-layers/src/hexagon-layer/hexagon-layer.js

```diff
@@ -85,6 +93,21 @@ export default class HexagonLayer extends CompositeLayer {
     }
   }
 
+  updateGetValueFuncs(props) {
+    let {getColorValue, getElevationValue} = props;
+    const {colorAggregation, getColorWeight, elevationAggregation, getElevationWeight} = this.props;
+
+    // If a custom `getColorValue` is provided (other than default) use it
+    if (getColorValue === defaultProps.getColorValue.value) {
+      getColorValue = getValueFunc(colorAggregation, getColorWeight);
```

Comment:
This is run on every layer update, and `getValueFunc` creates new functions whenever called. Check if the relevant props have changed before updating the state.

Reply:
done.